cl/scrapers/management/commands/scrape_caselaw.py

```diff
@@ -0,0 +1,339 @@
+import json
+import os
+from pathlib import Path
+from typing import TypedDict
+
+import requests
+from django.conf import settings
+
+from cl.lib.command_utils import VerboseCommand, logger
+from cl.settings import CASELAW_API_KEY
+
+reporters_slugs = dict()
+
+
+class OptionsType(TypedDict):
+    type: str
+    cursor: str
+    court: str
+    jurisdiction: str
+    last_updated: str
+    last_updated_filter: str
+    ordering: str
+    body_format: str
+    full_case: bool
+    reporter_id: int
+    source: str
+
+
+def get_data(url, params=None):
+    header = {"Authorization": f"Token {CASELAW_API_KEY}"}
+    logger.info(f"Processing endpoint: {url}")
+    try:
+        request = requests.get(url, headers=header, params=params, timeout=10)
+
+        # print(request.url)
+
+        if request.status_code == 401:
+            logger.error("Invalid token header or No credentials provided")
+            return None
+        if request.status_code == 404:
+            logger.error("Endpoint doesn't exist")
+            return None
+        if request.status_code == 500:
+            logger.error("Case.law server error")
+            return None
+
+        return request.json()
+
+    except requests.exceptions.ConnectionError:
+        return None
+
+    except requests.exceptions.ReadTimeout:
+        logger.error("Case.law server timeout, url: ", url)
+        return None
+
+
+def get_reporter_slug(reporter_endpoint):
+    """
+    Get reporter slug from reporter endpoint
+    :param reporter_endpoint: Reporter instance endpoint
+    @return: reporter slug or None
+    """
```

Comment:
this doctoring doesn't follow our standard style.  I would check another file to get the specific style down.  not '@' in return and we keep the first line as a short explanation.